src/Compilers/CSharp/Test/Emit/Attributes/InternalsVisibleToAndStrongNameTests.cs

```diff
@@ -829,6 +829,87 @@ protected B(C o)
         }
 
+        [Fact]
+        public void CS0281Method()
+        {
+            var friendClass = CreateCompilation(@"
+using System.Runtime.CompilerServices;
+[ assembly: InternalsVisibleTo(""cs0281, PublicKey=00240000048000009400000006020000002400005253413100040000010001002b986f6b5ea5717d35c72d38561f413e267029efa9b5f107b9331d83df657381325b3a67b75812f63a9436ceccb49494de8f574f8e639d4d26c0fcf8b0e9a1a196b80b6f6ed053628d10d027e032df2ed1d60835e5f47d32c9ef6da10d0366a319573362c821b5f8fa5abc5bb22241de6f666a85d82d6ba8c3090d01636bd2bb"") ]
+public class PublicClass
+{
+
+    internal static void InternalMethod() { }
+    protected static void ProtectedMethod() { }
+    private static void PrivateMethod() { }
+    internal protected static void InternalProtectedMethod() { }
+    private protected static void PrivateProtectedMethod() { }
+}", assemblyName: "Paul");
+
+            string cs0281 = @"
+
+public class Test
+{
+	static void Main ()
+	{
+		PublicClass.InternalMethod();
+        PublicClass.ProtectedMethod();
+        PublicClass.PrivateMethod();
+        PublicClass.InternalProtectedMethod();
+        PublicClass.PrivateProtectedMethod();
+	}
+}";
+            var other = CreateCompilation(cs0281, references: new[] { friendClass.EmitToImageReference() }, assemblyName: "cs0281", options: TestOptions.ReleaseDll.WithStrongNameProvider(s_defaultDesktopProvider));
+            other.VerifyDiagnostics(
+                      // (7,15): error CS0281: Friend access was granted by 'Paul, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null', but the public key of the output assembly ('') does not match that specified by the InternalsVisibleTo attribute in the granting assembly.
+                      // 		PublicClass.InternalMethod();
+                       Diagnostic(ErrorCode.ERR_FriendRefNotEqualToThis, "InternalMethod").WithArguments("Paul, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null", "").WithLocation(7, 15),
+                      // (8,21): error CS0122: 'PublicClass.ProtectedMethod()' is inaccessible due to its protection level
+                      //         PublicClass.ProtectedMethod();
+                      Diagnostic(ErrorCode.ERR_BadAccess, "ProtectedMethod").WithArguments("PublicClass.ProtectedMethod()").WithLocation(8, 21),
+                      // (9,21): error CS0117: 'PublicClass' does not contain a definition for 'PrivateMethod'
+                      //         PublicClass.PrivateMethod();
+                      Diagnostic(ErrorCode.ERR_NoSuchMember, "PrivateMethod").WithArguments("PublicClass", "PrivateMethod").WithLocation(9, 21),
+                      // (10,21): error CS0122: 'PublicClass.InternalProtectedMethod()' is inaccessible due to its protection level
+                      //         PublicClass.InternalProtectedMethod();
+                      Diagnostic(ErrorCode.ERR_BadAccess, "InternalProtectedMethod").WithArguments("PublicClass.InternalProtectedMethod()").WithLocation(10, 21),
```

Comment:
I think these two are the wrong message. "protected internal" and "private protected" should be the same as "internal"